### ISSUE TITLE
VM operator cannot update cluster status

### ISSUE THREAD

**Issue by stress-t** [VictoriaMetrics/operator#253]
No error config

```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMCluster
metadata:
  name: example-vmcluster-persistent
spec:
  retentionPeriod: "4"
  replicationFactor: 2
  vminsert:
    replicaCount: 2
    image:
      repository: victoriametrics/vminsert
```

Configuration when vm operator has errors in log

```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMCluster
metadata:
  name: example-vmcluster-persistent
spec:
  retentionPeriod: "4"
  replicationFactor: 2
  vminsert:
    replicaCount: 2
    image:
      repository: victoriametrics/vminsert
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: vminsert
              operator: In
              values:
              - "true"
```

Error logs
`{"level":"error","ts":1623077926.4322712,"logger":"factory","msg":"cannot update cluster status","error":"Operation cannot be fulfilled on vmclusters.operator.victoriametrics.com \"example-vmcluster-persistent\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tgithub.com/go-logr/zapr@v0.2.0/zapr.go:132\ngithub.com/VictoriaMetrics/operator/controllers/factory.CreateOrUpdateVMCluster.func1\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/vmcluster.go:63\ngithub.com/VictoriaMetrics/operator/controllers/factory.CreateOrUpdateVMCluster\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/vmcluster.go:210\ngithub.com/VictoriaMetrics/operator/controllers.(*VMClusterReconciler).Reconcile\n\tgithub.com/VictoriaMetrics/operator/controllers/vmcluster_controller.go:67\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:297\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:252\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\tsigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:215\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99"}`

```
apiVersion: v1
appVersion: 0.14.2
description: Victoria Metrics Operator
home: https://github.com/VictoriaMetrics/operator
name: victoria-metrics-operator
sources:
- https://github.com/VictoriaMetrics/operator
version: 0.1.13
```

K8s 1.20.7

**Comment by f41gh7** [comment]
Changes was included to the `0.15.0` release

**Comment by f41gh7** [comment]
feel free to re-open, if it happens again.
